src/session/mod.rs

```diff
@@ -400,7 +409,9 @@ impl Session {
                 .get_mut()
                 .retain(|sender| sender.send(file.clone()).is_ok());
 
-            if file.local.is_downloading_completed || entry.get().is_empty() {
+            if file.local.is_downloading_active == false | file.local.is_downloading_completed
```

Comment:
```suggestion
            if !file.local.is_downloading_active
```

If the downloading is completed then `is_downloading_active` is `false`, so this is enough.

Reply:
@YuraIz I meant to also remove the `file.local.is_downloading_completed` condition here.